client/src/components/LogInModal.jsx

```diff
@@ -0,0 +1,94 @@
+//Import necessary functionalities
+import React from 'react';
+import { UserStatus } from '../context/UserContext.jsx';
+import { useNavigate } from 'react-router-dom';
+
+const LogInModal = ({ isOpen, onClose }) => {
+
+    if (!isOpen) return null;
+
+    //Import display and user settings to use in display control and user updates
+    const { loggedUser, setLoggedUser } = UserStatus();
+    
+    //Call useNavigate hook to change component display upon log in
+    const navigate = useNavigate();
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+        
+        //Check that correct value is being submitted right now
+        console.log(event.currentTarget.username.value);
+
+        //Set the value of the username input field to be stored in username variable
+        const username = event.currentTarget.username.value;
+
+        //Future Plans: Add in password functionality - OAuth?
+
+        //If on submission, there is a username submitted, send it to the server to check its existence
+        if (username) {
+            try {
+                const response = await fetch('/api/users/', {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({ username })
+                });
+
+                if (!response.ok) {
+                    throw new Error("Failed to check log in");
+                }
+
+                const userStatus = await response.json();
+
+                //If there is a user with this username, log them in
+                if (userStatus.exists) {
+
+                    //Switch to dashboard display
+                    navigate('/dashboard');
+
+                    //Set the logged user state to the user id we will need to get the correct data
```

Comment:
I always feel like it's better to set the LoggedUser first then navigate to the next step.

Reply:
That makes sense! I updated that function with commit [f2292ce](https://github.com/tati-guzman/click-and-treat/pull/3/commits/f2292ce3d20c5b1f4dd651496a20fddafa8d1d3b) and switched the order. Thank you!